pycheribuild/config/compilation_targets.py

```diff
@@ -784,6 +784,59 @@ def uses_softfloat_by_default(cls, xtarget: "CrossCompileTarget"):
         return xtarget.is_riscv(include_purecap=True)
 
 
+class Sel4TargetInfo(BaremetalClangTargetInfo):
+    shortname: str = "sel4"
+
+    @property
+    def cmake_system_name(self) -> str:
+        return "sel4"
```

Comment:
Is this actually a recognized value? If not you should use Generic so that CMake does the right thing.

Reply:
It doesn't do any weird things for me, I get all the expected *sel4* directories and symlinks. What does Generic do?

Reply:
CMake will use the default baremetal settings (https://github.com/Kitware/CMake/blob/master/Modules/Platform/Generic.cmake) with "Generic", I can't see a file called sel4.cmake, so my guess would be that it will do the wrong thing.

See https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html#toolchain-files:
`If your target is an embedded system without an OS, set CMAKE_SYSTEM_NAME to “Generic.” `